grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ng.html

```diff
@@ -1,34 +1,31 @@
-<div *ngIf="flowRunning$ | async" class="in-progress">
-  {{ (args$ | async).query }}
+<div *ngIf="flowError$ | async; else noError" class="error">
+  An error was encountered.
+  <div *ngIf="resultsStderr$ | async as resultsStderr">
+    stderr is: {{resultsStderr}}
+  </div>
 </div>
 
-<div *ngIf="flowCompleted$ | async" class='results'>
-  <div *ngIf="resultTable$ | async as resultTable">
-    <div class="results-query">
-      <span class="results-query-label">Query:</span>
-      <span class="results-query-text">{{resultTable.query}}</span>
-    </div>
-
-    <div class="table-container">
-      <table>
-        <thead>
-          <th *ngFor="let column of resultTable.header.columns; trackBy: trackByIndex">{{column.name}}</th>
-        </thead>
-        <tbody>
-          <tr *ngFor="let row of resultTable.rows; trackBy: trackByIndex">
-            <td *ngFor="let cellValue of row.values">
-              {{cellValue}}
-            </td>
-          </tr>
-        </tbody>
-      </table>
-    </div>
+<ng-template #noError>
+  <div *ngIf="flowRunning$ | async" class="in-progress">
+    Query in progress: {{ (args$ | async).query }}
   </div>
-</div>
 
-<div *ngIf="flowError$ | async" class="error">
-  An error was encountered.
-  <div *ngIf="resultStderr$ | async as resultStderr">
-    stderr is: {{resultStderr}}
+  <div *ngIf="resultsTable$ | async as resultsTable; else noResultsTable" class="results-table">
+    <osquery-results-table
+      [tableToDisplay]="resultsTable">
+    </osquery-results-table>
   </div>
-</div>
+
+  <ng-template #noResultsTable>
+    <div *ngIf="progressTable$ | async as progressTable" class="progress-table">
+      <div *ngIf="additionalRowsAvailable$ | async as additionalRowsAvailable" class="show-additional">
+        <span class="show-additional-text">{{additionalRowsAvailable}} row(s) hidden.</span>
```

Comment:
Use [`i18nPlural`](https://angular.io/api/common/I18nPluralPipe) instead of `row(s)`.